src/software/ai/motion_constraint/motion_constraint_visitor.cpp

```diff
@@ -34,36 +28,44 @@ void MotionConstraintVisitor::visit(const PivotKickTactic &tactic) {}
 
 void MotionConstraintVisitor::visit(const KickoffChipTactic &tactic)
 {
-    current_allowed_constraints = std::set<TbotsProto::MotionConstraint>({
-        TbotsProto::MotionConstraint::CENTER_CIRCLE,
-        TbotsProto::MotionConstraint::HALF_METER_AROUND_BALL
-    });
+    current_motion_constraints.erase(TbotsProto::MotionConstraint::CENTER_CIRCLE);
+    current_motion_constraints.erase(TbotsProto::MotionConstraint::ENEMY_HALF);
+    current_motion_constraints.erase(
+        TbotsProto::MotionConstraint::HALF_METER_AROUND_BALL);
+}
```

Comment:
Should we be inserting `ENEMY_HALF_WITHOUT_CENTRE_CIRCLE` here? I'm not sure what our desired behaviour is, but currently the robot is able to enter the enemy's half (excluding centre circle) and kick the ball. @itsarune Do you know if this is okay or if we should be staying in our own half? Or is this just an edge case that we shouldn't be worried about?

https://user-images.githubusercontent.com/28585597/219842747-672314e3-6b97-4b72-9a1c-78d7d7823d6c.mp4

Reply:
Sure, I inserted `ENEMY_HALF_WITHOUT_CENTRE_CIRCLE` for `KickoffChipTactic`, so this should cover the edge case when the ball is placed outside of the centre circle.